subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/ComponentAttributesRulesIntegrationTest.groovy

```diff
@@ -35,17 +35,29 @@ class ComponentAttributesRulesIntegrationTest extends AbstractModuleDependencyRe
             configurations {
                 conf.attributes.attribute(quality, 'qa')
             }
+
+            class AttributeRule implements ComponentMetadataRule {
+                Attribute targetAttribute
```

Comment:
This is an interesting case for caching: the type of the rule parameter is not snapshottable/isolatable. We'll have to do something.

Reply:
Changed to make `Attribute` snapshottable / isolatable.